lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -233,8 +233,12 @@ module InternalConstants
     Fluent::Plugin.register_output('google_cloud', self)
 
     PLUGIN_NAME = 'Fluentd Google Cloud Logging plugin'.freeze
-    # Extract plugin version by finding the spec this file was loaded from.
     PLUGIN_VERSION = begin
+      # Extract plugin version from file path.
+      __FILE__.match(
+        %r{fluent-plugin-google-cloud-(?<version>[0-9a-zA-Z\.]*)/})['version']
+    rescue StandardError
```

Comment:
This seems dangerous — there are many subclasses of `StandardError`... Why not:
```ruby
    PLUGIN_VERSION = begin
      # Extract plugin version from file path.
      match_data = __FILE__.match(
        %r{fluent-plugin-google-cloud-(?<version>[0-9a-zA-Z\.]*)/})
      if match_data
        match_data['version']
      else
        # Extract plugin version by finding the spec this file was loaded from.
        dependency = Gem::Dependency.new('fluent-plugin-google-cloud')
        all_specs, = Gem::SpecFetcher.fetcher.spec_for_dependency(dependency)
        matching_spec, = all_specs.grep(
           proc { |spec,| __FILE__.include?(spec.full_gem_path) }) do |spec,|
             spec.version.to_s
           end
        matching_spec
      end
    end.freeze
```
?

We could also extract this code into a private function, and then use an early `return` instead:
```ruby
    def extract_plugin_version
      # Extract plugin version from file path.
      match_data = __FILE__.match(
        %r{fluent-plugin-google-cloud-(?<version>[0-9a-zA-Z\.]*)/})
      return match_data['version'] if match_data
      # Extract plugin version by finding the spec this file was loaded from.
      dependency = Gem::Dependency.new('fluent-plugin-google-cloud')
      all_specs, = Gem::SpecFetcher.fetcher.spec_for_dependency(dependency)
      matching_spec, = all_specs.grep(
         proc { |spec,| __FILE__.include?(spec.full_gem_path) }) do |spec,|
           spec.version.to_s
         end
      matching_spec
    end
    PLUGIN_VERSION = extract_plugin_version.freeze
```